person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(id, age, name = 'Unknown', parent_permission: true)
+    @id = id
```

Comment:
Hi @firepicaso, 

Nice work progressing through this project! 🚀 You've laid down a solid foundation with your `Person` class. 🏗️ Your understanding of OOP concepts is evident through your method and properties utilization. However, there are a few aspects that need adjusting to meet the project's requirements more precisely. 

## Points to address: 🔍👨‍🔧

1. **ID Management**:
   - The project specification requires `@id` to be generated within the class, ensuring a unique identifier for each instance. This can be achieved by generating a random number, as shown in the requirements.
     ```ruby
     @id = Random.rand(1..1000)
     ```
   Remember, this means `id` should not be a parameter in your `initialize` method, as it's supposed to be generated, not passed in.

2. **Constructor Parameters**:
   - The `initialize` method should accept `age`, `name`, and `parent_permission` as parameters. Note that `name` and `parent_permission` should have default values of 'Unknown' and `true`, respectively.
     ```ruby
     def initialize(age, name = "Unknown", parent_permission = true)
     ```
   
3. **Attribute Accessibility**:
   - You've done well implementing `attr_accessor` and `attr_reader`! Just ensure you don't allow writing for `:id` as per the requirements, which you've adhered to. 🛡️👍 

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me, so I receive your feedback and queries promptly.

-------
As outlined in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you believe the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).